bids/grabbids/bids_layout.py

```diff
@@ -63,10 +49,10 @@ def get_metadata(self, path):
     def get_fieldmap(self, path):
         sub = os.path.split(path)[1].split("_")[0].split("sub-")[1]
         fieldmap_set = {}
-        for file in self.get(subject=sub,
-                             type='(phase1|phase2|phasediff|epi|fieldmap)',
+        type_ = '(phase1|phase2|phasediff|epi|fieldmap)'
+        for file in self.get(subject=sub, type=type_,
                              extensions=['nii.gz', 'nii']):
-            metadata = self.get_metadata(file.filename)
+            metadata = self.get_metadata(file.filename, type=type_)
```

Comment:
I'm pretty sure if you remove that, the tests will fail because you'll get back files that are above the target file in the hierarchy, and match the subject, but have types other than phase1, phase2, etc.